### ISSUE TITLE
Finding a whitelist

### ISSUE THREAD

**Issue by NicolasHipp** [Hoohm/CITE-seq-Count#48]
Hi everyone, 

I'm trying to demultiplex my scRNAseq data with CITE-seq-count (hashtag barcoding)

I tested with 10 000 reads
`CITE-seq-Count -R1 HTO-1_R1.fastq.gz -R2 HTO-1_R2.fastq.gz -t hto.csv -cbf 1 -cbl 16 -umif 17 -umil 26 -o /Results -cells 5000 -n 10000 `
which worked pretty well..

So I discard the -n 10000 in order to used all the reads 
```
CITE-seq-Count -R1 HTO-1_R1.fastq.gz -R2 HTO-1_R2.fastq.gz -t hto.csv -cbf 1 -cbl 16 -umif 17 -umil 26 -o /Results -cells 5000 
Counting number of reads
Started mapping
Processing 41,301,352 reads
CITE-seq-Count is running with 72 cores.
...
Mapping done
Merging results
Correcting cell barcodes
Finding a whitelist
/.local/lib/python3.7/site-packages/umi_tools/whitelist_methods.py:283: RuntimeWarning: invalid value encountered in sqrt
  lineVecNorm = lineVec / np.sqrt(np.sum(lineVec**2))
Traceback (most recent call last):
  File "/.local/bin/CITE-seq-Count", line 10, in <module>
    sys.exit(main())
  File "/.local/lib/python3.7/site-packages/cite_seq_count/__main__.py", line 352, in main
    collapsing_threshold=args.bc_threshold)
  File "/.local/lib/python3.7/site-packages/cite_seq_count/processing.py", line 310, in correct_cells
    plotfile_prefix=False)
  File "/.local/lib/python3.7/site-packages/umi_tools/whitelist_methods.py", line 447, in getCellWhitelist
    cell_barcode_counts, cell_number, plotfile_prefix)
  File "/.local/lib/python3.7/site-packages/umi_tools/whitelist_methods.py", line 322, in getKneeEstimateDistance
    raise ValueError("Something's gone wrong here!!")
ValueError: Something's gone wrong here!!
```

Is there something that I missed? 
Thanks!!
nicolas

**Comment by Hoohm** [comment]
Hello @NicolasHipp 
Other users have reported the same issue.
Using the default function of the new umi_tools (1.0.0) introduced a new way of calculating the whitelist.

This is where the error comes from. It doesn't happen with all the datasets and it slipped by my tests.

I'm trying to fix it and have already pushed a few changes on the develop branch. Could you test it on your dataset?

Best wishes

**Comment by NicolasHipp** [comment]
Woo It seems to work well with the developing branch !! 
![image](https://user-images.githubusercontent.com/18481923/54355756-331c3b80-465a-11e9-93fb-df0d83f44468.png)
Thanks a lots, If you want some informations about the sequencing or anything else to test your soft telle me! 
Now I have just to figure out why I have cells which seems strongly positive for HTO1 and have few reads in HTO2 or inversely true

Best 
Nicolas

**Comment by Hoohm** [comment]
Great news!
Have you tried using Seurat for HTO dumxing?

**Comment by NicolasHipp** [comment]
Nop  this is the next step, I received datas from HTO before 5' so I just checked if the HTO sequencing worked!
Best,
Nico

**Comment by hothri** [comment]
Hi, 
I am getting similar error while using the current verison of CITEseq count (v1.4.2) and umitools (v1.0.0) 

/CITE_seq_tool/cite_env/lib/python3.6/site-packages/umi_tools/whitelist_methods.py:283: RuntimeWarning: invalid value encountered in sqrt
  lineVecNorm = lineVec / np.sqrt(np.sum(lineVec**2))
Mapping done
Merging results
Correcting cell barcodes
Finding a whitelist
Traceback (most recent call last):
  File "/CITE_seq_tool/cite_env/bin/CITE-seq-Count", line 10, in <module>
    sys.exit(main())
  File "/CITE_seq_tool/cite_env/lib/python3.6/site-packages/cite_seq_count/__main__.py", line 352, in main
    collapsing_threshold=args.bc_threshold)
  File "/CITE_seq_tool/cite_env/lib/python3.6/site-packages/cite_seq_count/processing.py", line 310, in correct_cells
    plotfile_prefix=False)
  File "/CITE_seq_tool/cite_env/lib/python3.6/site-packages/umi_tools/whitelist_methods.py", line 447, in getCellWhitelist
    cell_barcode_counts, cell_number, plotfile_prefix)
  File "/CITE_seq_tool/cite_env/lib/python3.6/site-packages/umi_tools/whitelist_methods.py", line 322, in getKneeEstimateDistance
    raise ValueError("Something's gone wrong here!!")
ValueError: Something's gone wrong here!!

How did you fix this?? 
Thank you, 
Hothri

**Comment by Hoohm** [comment]
Hello @hothri 
This bug is fixed on the develop branch.
I'm wrapping things up, it should come out soon as a new release.

**Comment by Hoohm** [comment]
Should be fixed in the new 1.4.3 release. Closing this.
